cluster-autoscaler/core/scale_down.go

```diff
@@ -572,6 +572,46 @@ func (sd *ScaleDown) mapNodesToStatusScaleDownNodes(nodes []*apiv1.Node, nodeGro
 	return result
 }
 
+// SoftTaintUnneededNodes manage soft taints of unneeded nodes.
+func (sd *ScaleDown) SoftTaintUnneededNodes(allNodes []*apiv1.Node) (errors []error) {
+	apiCallBudget := sd.context.AutoscalingOptions.MaxBulkSoftTaint
+	// Check if some new nodes should be tainted
+	for _, node := range allNodes {
+		if deletetaint.HasToBeDeletedTaint(node) {
+			// Do not consider nodes that are scheduled to be deleted
+			continue
+		}
+		alreadyTainted := deletetaint.HasDeletionCandidateTaint(node)
+		_, unneeded := sd.unneededNodes[node.Name]
+
+		if unneeded != alreadyTainted {
+			apiCallBudget--
+			if apiCallBudget < 0 {
+				continue
+			}
+			if unneeded {
+				// Node unneeded but not tainted
```

Comment:
If you follow my suggestion above you can drop this comment and one below :)

Reply:
I had that previously, I have refactored into current form to account for apiCallBudget.